src/solver/mha/mha_ck_fa_v2_solver_forward.cpp

```diff
@@ -0,0 +1,268 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/mha/solvers.hpp>
+#include <miopen/mha/invoke_params.hpp>
+#include <miopen/datatype.hpp>
+#include <miopen/hipoc_kernel.hpp>
+
+#if MIOPEN_USE_COMPOSABLEKERNEL
+#include "ck_tile/ops/fmha.hpp"
+#include "ck_tile/ops/fmha_fwd.hpp"
+#include "ck/stream_config.hpp"
+#endif
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_DEBUG_FA_CK_V2_FWD)
+
+namespace miopen {
+
+namespace solver {
+
+namespace mha {
+
+#if MIOPEN_USE_COMPOSABLEKERNEL
+static std::string Convert(miopenDataType_t dataType)
+{
+    switch(dataType)
+    {
+    case miopenHalf: {
+        return "fp16";
+    }
+    case miopenBFloat16: {
+        return "bfp16";
+    }
+    default: {
+        MIOPEN_THROW("Unsupported datatype provided");
+    }
+    }
+}
+#endif
+
+bool MhaCKFlashAttentionV2Forward::IsApplicable(
+    [[maybe_unused]] const ExecutionContext& context,
+    const miopen::mha::ProblemDescription& problem) const
+{
+#if MIOPEN_USE_COMPOSABLEKERNEL
+    if(!problem.IsForward())
+    {
+        return false;
+    }
+
+    auto deviceName = context.GetStream().GetDeviceName();
+    if(!StartsWith(deviceName, "gfx94") && deviceName != "gfx90a")
+    {
+        return false;
+    }
+
+    auto& descsFwd            = problem.GetDescsForward();
```

Comment:
nitpick `const`

Reply:
I like adding it as well to be explicit (and will), but in this case they are equivalent.